File: articles/multi-criteria-decision-analysis/python/run_all_mcda_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Multi-Criteria Decision Analysis."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "weighted_score_model.py",
    "criteria_normalization.py",
    "rank_stability_analysis.py",
    "stakeholder_weight_profiles.py",
    "outranking_review.py",
    "mcda_review_queue.py",
    "decision_record_exporter.py",
    "mcda_weight_sensitivity_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python MCDA workflows completed.")


if __name__ == "__main__":
    main()

File: articles/multi-criteria-decision-analysis/r/run_all_mcda_workflows.R
# run_all_mcda_workflows.R
# Run all R workflows for Multi-Criteria Decision Analysis.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "mcda_workflow.R",
  "weighted_score_tables.R",
  "rank_stability_tables.R",
  "sensitivity_review_tables.R",
  "criteria_contribution_reports.R",
  "mcda_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R MCDA workflows completed.")
